docs/how_to/use_as_tsdb.md

```diff
@@ -0,0 +1,155 @@
+
+For example, continuing with the vehicle fleet tracking example introduced earlier, a schema might look as follows:
+
+```protobuf
+syntax = "proto3";
+
+message VehicleLocation {
+  double latitude = 1;
+  double longitude = 2;
+  double fuel_percent = 3;
+  string status = 4;
+}
+```
+
+While M3DB strives to support the entire [proto3 language spec](https://developers.google.com/protocol-buffers/docs/proto3), only the following features are currently supported:
+
+1. [Scalar values](https://developers.google.com/protocol-buffers/docs/proto3#scalar)
+2. Nested messages
+3. Repeated fields
+4. Map fields
+5. Reserved fields
+
+The following features are currently not supported:
+
+1. `Any` fields
+2. [`Oneof` fields](https://developers.google.com/protocol-buffers/docs/proto#oneof)
+3. Options of any type
+4. Custom field types
+
+#### Compression
+
+While M3DB supports schemas that contain nested messages, repeated fields, and map fields, currently it can only effectively compress top level scalar fields. For example, M3DB can compress every field in the following schema:
+
+```protobuf
+syntax = "proto3";
+
+message VehicleLocation {
+  double latitude = 1;
+  double longitude = 2;
+  double fuel_percent = 3;
+  string status = 4;
+}
+```
+
+however, it will not apply any form of compression to the `attributes` field in this schema:
+
+```protobuf
+syntax = "proto3";
+
+message VehicleLocation {
+  double latitude = 1;
+  double longitude = 2;
+  double fuel_percent = 3;
+  string status = 4;
+  map<string, string> attributes = 5;
+}
+```
+
+This isn't to say that the schema above shouldn't be used, but users should take into account that any data stored in the `attributes` map will be held in memory and eventually stored on disk with no compression.
+
+For more details on the compression scheme and its limitations, review [the documentation for M3DB's compressed protobuf encoding](https://github.com/m3db/m3/blob/master/src/dbnode/encoding/proto/docs/encoding.md).
+
+
+### Getting Started
+
+#### M3DB setup
+
+For more advanced setups, its best to follow the guide's on how to configure an [M3DB cluster manually](./cluster_hard_way.md) or [using Kubernetes](./kubernetes.md). However, this tutorial will walk you through configuring a single node setup locally for development.
+
+First, run the following command to pull the latest M3DB image:
+
+```
+docker pull quay.io/m3db/m3dbnode:latest
+```
+
+Next, run the following command to start the M3DB container:
+
+```
+docker run -p 7201:7201 -p 7203:7203 -p 9000:9000 -p 9001:9001 -p 9002:9002 -p 9003:9003 -p 9004:9004 -p 2379:2379 --name m3db -v $(pwd)/m3db_data:/var/lib/m3db -v $(pwd)/src/dbnode/config/m3dbnode-local-etcd-proto.yml:/etc/m3dbnode/m3dbnode.yml -v <PATH_TO_SCHEMA_PROTOBUF_FILE>:/etc/m3dbnode/default_schema.proto quay.io/m3db/m3dbnode:latest
+```
+
+Breaking that down:
```

Comment:
do we need to explain how docker works? can we just say "See `docker run --help` for more information."?

it might be better to just call out the specific flag values, e.g.:

- Creating a bind mount (`$(pwd)/m3db_data:/var/lib/m3db`) enables M3DB data to persist between container restarts.
- Mounting the configuration file (`m3dbnode.yml`) enables configuration updates by restarting the container (rather than rebuilding it). [other info about the config file]
- Similar to configuration, mounting the schema file (`default_schema.proto`) enables schema updates by restarting the container. [other info about the schema file]

Reply:
I prefer to err on the side of verbosity here a little bit, but some of your clarifications are good I'll add them